### ISSUE TITLE
Japanese wordlist sorted?

### ISSUE THREAD

**Issue by wizardofozzie** [trezor/python-mnemonic#26]
From [my SO question](https://stackoverflow.com/questions/31156424/binary-search-bisect-for-a-unicode-character-in-python-2-7), it appears that the Japanese wordlist *is not sorted*, as is required for the `binary_search` function.

**Comment by voisine** [comment]
I don't think we can change it now. Once it's published, people may have already started using it. Does japanese have a standard alphabetic order? Or would it just be based on the unicode character values? Or maybe the UTF-8 representation?

**Comment by simcity4242** [comment]
@voisine >*Does Japanese have a standard alphabetical order?*

No idea. I'm just going by what is said in the [StackOverflow post of mine](https://stackoverflow.com/questions/31156424/binary-search-bisect-for-a-unicode-character-in-python-2-7) as the `binary_search` function is part of [this repository's source code](https://github.com/trezor/python-mnemonic/blob/master/mnemonic/mnemonic.py#L38-L42), afterall, so as the [SO response notes](http://stackoverflow.com/a/31156743/3936486), **binary search functions require _sorted_ word lists**. As such, [Japanese BIP39 test vectors fail](https://github.com/bip32JP/bip32JP.github.io/blob/master/test_JP_BIP39.json) for *some* of the vectors if `binary_search` is used. 

As you say the wordlist order is cumbersome to change; therefore the binary_search function must be deprecated in favour of a traditional (and slower) `list.index`

**Comment by simcity4242** [comment]
@voisine Let me rephrase: what purpose does `binary_search` serve if it needs a sorted list? English words are fine: they're sorted.

It appears to be an artefact of `to_entropy`, since all other functions use list index. I *believe* this is why it hasn't been caught yet: has anyone tried going back to entropy from a valid Japanese mnemonic?

1. The `binary_search` won't raise an index exception, it'll return -1, which isn't caught either
2. The indexing will be **completely wrong** for ~50% of the words. 

Is this something to bring to BIPS, Trezor, this repo?

**Comment by prusnak** [comment]
I am not sure if anything could be done now. Closing.
